app/crud/geostore.py

```diff
@@ -0,0 +1,133 @@
+import json
+from typing import List
+from uuid import UUID
+
+from asyncpg.exceptions import UniqueViolationError
+from geojson import Feature as geoFeature
+from geojson import FeatureCollection as geoFeatureCollection
+
+from app.application import db
+from app.errors import BadRequestError, RecordNotFoundError
+from app.models.orm.user_areas import UserArea as ORMUserArea
+from app.models.pydantic.geostore import Feature, Geometry, Geostore, GeostoreHydrated
+
+
+async def get_geostore_from_anywhere(geostore_id: UUID) -> GeostoreHydrated:
+    sql = db.text(
+        """
+        SELECT *
+        FROM geostore
+        WHERE gfw_geostore_id=:geostore_id;
+        """
+    )
+    bind_vals = {"geostore_id": f"{geostore_id}"}
+    sql = sql.bindparams(**bind_vals)
+
+    row = await db.first(sql)
+
+    if row is None:
+        raise RecordNotFoundError(
+            f"Area with gfw_geostore_id {geostore_id} does not exist"
+        )
+
+    geo: Geostore = Geostore.from_orm(row)
+
+    return hydrate_geostore(geo)
+
+
+async def get_geostore_by_version(dataset, version, geostore_id) -> GeostoreHydrated:
+    sql = db.text(
+        f"""
+        SELECT *
+        FROM ONLY "{dataset}"."{version}"
```

Comment:
I believe `ONLY` means here to exclude descendant tables. We might need to partition some of these tables and hence there will be descendant tables. These tables also have more than the standard geostore columns. I would not use the wildcard `*` but list the fields explicitly. 

To assure that you always use the same fields names across the different queries you can use SQL alchemy core.

```python
columns: List[Column] = [db.column(gfw_geostore_id), db.column.(gfw_geojson), db.column(gfw_bbox), db.column(gfw.area__ha)]

src_table: Table = db.table(version)
src_table.schema = dataset

where_clause: TextClause = db.text("gfw_geostore_id=:geostore_id") 
bind_vals = {"geostore_id": f"{geostore_id}"}
where_clause  = where_clause.bindparams(**bind_vals)

sql: Select = db.select(columns).select_from(src_table).where(where_clause)
```

Reply:
D'oh. Right you are. Thanks for the SA Core lesson. Changed and pushed.